pkg/reconciler/knativeserving/knativeserving.go

```diff
@@ -88,8 +88,10 @@ func (r *Reconciler) FinalizeKind(ctx context.Context, original *v1alpha1.Knativ
 // converge the two.
 func (r *Reconciler) ReconcileKind(ctx context.Context, ks *v1alpha1.KnativeServing) pkgreconciler.Event {
 	logger := logging.FromContext(ctx)
-	ks.Status.InitializeConditions()
-	ks.Status.ObservedGeneration = ks.Generation
+	if len(ks.Status.Conditions) == 0 {
+		ks.Status.InitializeConditions()
+		ks.Status.ObservedGeneration = ks.Generation
```

Comment:
We definitely want to run this on every reconcile as we want to make sure that `ObservedGeneration` is consistently updated.
Are we only churning through `resourceVersion` or also through `generation`? The latter would indicate we're constantly updating the Spec.